File: generated-code/fsharp/codecraft/FileReadWrite/Model/BuildAction.fs
#nowarn "0058"

namespace TransGenTest.Model

open TransGenTest

/// Build action
type BuildAction = {
    /// Type of an entity to build
    EntityType: Model.EntityType;
    /// Desired position of new entity
    Position: Vec2Int;
} with

    /// Write BuildAction to writer
    member this.writeTo(writer: System.IO.BinaryWriter) =
        writer.Write (int this.EntityType)
        this.Position.writeTo writer

    /// Read BuildAction from reader
    static member readFrom(reader: System.IO.BinaryReader) = {
        EntityType = reader.ReadInt32() |> enum
        Position = Vec2Int.readFrom reader;
    }

File: generated-code/typescript/codecraft/FileReadWrite/src/model/build-action.ts
import { EntityType } from "./entity-type";
import { Vec2Int } from "../vec2-int";
import { Stream } from "../stream";

/**
 * Build action
 */
export class BuildAction {
    /**
     * Type of an entity to build
     */
    entityType: EntityType
    /**
     * Desired position of new entity
     */
    position: Vec2Int

    constructor(entityType: EntityType, position: Vec2Int) {
        this.entityType = entityType;
        this.position = position;
    }

    /**
     * Read BuildAction from input stream
     */
    static async readFrom(stream: Stream): Promise<BuildAction> {
        let entityType;
        entityType = await EntityType.readFrom(stream);
        let position;
        position = await Vec2Int.readFrom(stream);
        return new BuildAction(entityType, position)
    }

    /**
     * Write BuildAction to output stream
     */
    async writeTo(stream: Stream) {
        let entityType = this.entityType;
        await entityType.writeTo(stream);
        let position = this.position;
        await position.writeTo(stream);
    }
}